octoprint_bambu_printer/printer/bambu_virtual_printer.py

```diff
@@ -359,16 +366,46 @@ def _sd_status(self) -> bool:
 
     @gcode_executor.register("M23")
     def _select_sd_file(self, data: str) -> bool:
+        self._log.debug("M23 command received.")
         filename = data.split(maxsplit=1)[1].strip()
-        return self.select_project_file(filename)
+
+        # ** Step 1: Perform the deselection logic **
+        self._log.debug("Calling remove_project_selection as part of M23 handling.")
+        # Call the remove_project_selection method to clear any previous selection state
+        self.remove_project_selection()
+        # remove_project_selection will call _send_file_selected_message
+        # and send the explicit deselection messages over serial.
+
+        # Add a small delay here to allow OctoPrint's UI to potentially process
+        # the deselection messages before the selection messages arrive.
+        # This might help synchronize the UI state.
+        time.sleep(0.05) # Small delay (50 milliseconds)
+
+        # ** Step 2: Proceed with the original M23 selection logic **
+        self._log.debug(f"Proceeding with selection for filename: {filename}")
+        # Call the select_project_file method to set the new selection
+        # select_project_file will call _send_file_selected_message
+        # and send the selection messages for the new file.
+        success = self.select_project_file(filename)
+
+        # The "ok N+1" response for the M23 command is handled automatically
+        # by _process_gcode_serial_command after this method returns.
+
+        return success # Return whether the selection was successful
+
 
     def _send_file_selected_message(self):
         if self.selected_file is None:
             return
 
-        self.sendIO(f"File opened: {self.selected_file.dosname} Size: {self.selected_file.size}")
+        self.sendIO(
+            f"File opened: {self.selected_file.file_name}  "
```

Comment:
Sorry if I missed this before, but I'm pretty sure that using file_name here instead of dosname is what causes OctoPrint to not recognize the file that is being printed currently. Please explain the reasoning behind changing this?